### PR TITLE
add withdraw handles on ledger

### PR DESCRIPTION
**Pull request by EduardoRFS**
## Depends

- [x] #89

## Problem

With #88 the consensus contract allow the user to provide a proof that he burned the money on the sidechain, but currently the patricia tree is not integrated in the sidechain itself.

## Solution

This doesn't solve the problem entirely, but it creates the needed infrastructure from the ledger to integrate the patricia tree in the future on the protocol.

## Related

- #34 
- #88

### CODE REVIEW

**Review by anchpop — APPROVED** [review]

**Comment by anchpop** [comment]
Can you point me to the bit of code that does increments the ID? I can't seem to find it. Or do we now use the position in the patricia tree as the ID?

**Comment by EduardoRFS** [comment]
@anchpop https://github.com/marigold-dev/sidechain/blob/main/helpers/incremental_patricia.re#L86

TLDR if you're using Incremental_patricia the key is always incremented by one when adding.
